contracts/safe/SafeERC20.sol

```diff
@@ -0,0 +1,42 @@
+pragma solidity ^0.5.2;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+library SafeERC20 {
+    using SafeMath for uint256;
+
+    function transferTokens(
+      ERC20 _token,
+      address _from,
+      address _to,
+      uint256 _value
+    ) internal {
+        uint256 oldBalance = _token.balanceOf(_to);
+        require(
```

Comment:
Yep. Analogous to SafeMath, it validates that the token calls had the expected outcome.

It's worth debating whether or not we really need that. For trusted tokens (when we are confident the contract behaves in accordance with ERC20 as a specification) this is extra overhead. We ended up including these in our Payments contract as a hedge against anticipated audit feedback (based on similar commentary from an example audit).

It makes a bit more sense here, since users can call in with arbitrary token contracts. There's an argument to be made that, so long as we can come out with BEEs in the end, we don't care about the specifics of that token contract, but I'm inclined to be risk-averse at this point.

Reply:
👍